### PR TITLE
FIX if a language is set in subsites make sure this is used in

### PR DESCRIPTION
**Pull request by ohararyan**
See issue here https://github.com/silverstripe/silverstripe-subsites/issues/83

### CODE REVIEW

**Comment by wilr** [comment]
Ideally this would be refactored so that translatable, subsites and module X could all hook in. Also test failing so will need to investigate that

File: code/controllers/ContentController.php [hunk 1]
```diff
@@ -363,21 +363,25 @@ public function SiteConfig() {
...
 	 */
 	public function ContentLocale() {
 		if($this->dataRecord && $this->dataRecord->hasExtension('Translatable')) {
 			$locale = $this->dataRecord->Locale;
 		} elseif(class_exists('Translatable') && SiteTree::has_extension('Translatable')) {
 			$locale = Translatable::get_current_locale();
+		} elseif(class_exists('Subsite') && $subsite = Subsite::currentSubsite()) {
+			$locale = i18n::get_locale_from_lang($subsite->Language);
+			if($locale) {
+				i18n::set_locale($locale);
```
Comment (tractorcow) [inline code/controllers/ContentController.php]: Not very good style; A getter should not by nature modify the state.

**Comment by tractorcow** [comment]
If you want to allow extensions (such as subsites) set the locale, then the correct way to do it is via an extension.


```
public function ContentLocale(){
	$locale = ''; // some logic
	$this->extend('updateContentLocale', $locale);
	return i18n::convert_rfc1766($locale);
}
```

This means that we have less module-specific code, and other modules can also hook into this behaviour.

This would need to go into the 3 branch please. :D

File: code/controllers/ContentController.php [hunk 2]
```diff
@@ -363,21 +363,25 @@ public function SiteConfig() {
...
-	 * 
+	 *
 	 * Suitable for insertion into lang= and xml:lang=
 	 * attributes in HTML or XHTML output.
-	 * 
+	 *
 	 * @return string
 	 */
 	public function ContentLocale() {
 		if($this->dataRecord && $this->dataRecord->hasExtension('Translatable')) {
```
Comment (tractorcow) [inline code/controllers/ContentController.php]: Changing this to `$this->dataRecord->hasField('Locale')` would mean being able to remove the hard-coded translatable dependency too.

File: code/controllers/ContentController.php [hunk 3]
```diff
@@ -363,21 +363,25 @@ public function SiteConfig() {
...
 			$locale = $this->dataRecord->Locale;
 		} elseif(class_exists('Translatable') && SiteTree::has_extension('Translatable')) {
```
Comment (tractorcow) [inline code/controllers/ContentController.php]: To delete; Made redundant by the above check. :P

**Pull request closed without merging**